src/test/harness.rs

```diff
@@ -187,16 +183,26 @@ impl ExpectedMessage {
     }
 }
 
-pub fn expect_messages(results: LsResultList, expected: &[&ExpectedMessage]) {
-    let start_clock = SystemTime::now();
-    let mut results_count = results.lock().unwrap().len();
-    while results_count < expected.len() {
-        if start_clock.elapsed().unwrap().as_secs() >= TEST_TIMEOUT_IN_SEC {
-            panic!("Hit timeout");
+macro_rules! wait_for_n_results {
```

Comment:
Why use a macro here instead of a function?

Reply:
Macros in tests provide line reference to the failing point in the test case without having to enable and trawl though backtraces, ie
```
thread 'test::test_deglob' panicked at 'Timeout waiting for a result', src/test/mod.rs:1249:8
```

Reply:
Yikes. That is an interesting and surely unintended effect. Given that it is pretty easy for the user to get a backtrace, I'm not sure whether it is worth the hassle of using a macro though.

Reply:
I think it's perfectly intended, consider the usefulness of `assert` without it. Still I can replace this with a function if you prefer.